src/main/java/com/aws/greengrass/certificatemanager/certificate/ClientCertificateGenerator.java

```diff
@@ -44,19 +45,24 @@ public ClientCertificateGenerator(X500Name subject, PublicKey publicKey, Consume
         this.certificatesConfig = certificatesConfig;
     }
 
-    /**
-     * Regenerates certificate.
-     *
-     * @param connectivityInfoSupplier ConnectivityInfo Supplier (not used in this implementation)
-     * @throws KeyStoreException if unable to retrieve CA key/cert
-     */
     @Override
-    public synchronized void generateCertificate(Supplier<List<String>> connectivityInfoSupplier)
+    public synchronized void generateCertificate(Supplier<List<String>> connectivityInfoSupplier, String reason)
             throws KeyStoreException {
         Instant now = Instant.now(clock);
 
-        logger.atInfo().kv("subject", subject)
+        logger.atInfo()
+                .kv("subject", subject)
+                .kv("reason", reason)
+                .kv("previousCertExpiry",
+                        certificate == null
+                                ? "N/A" :
+                                getExpiryTime())
+                .kv("previousCertValidityHMS",
+                        certificate == null
+                                ? "N/A" :
+                                DurationFormatUtils.formatDurationHMS(getValidity().toMillis()))
```

Comment:
I'd just use toSeconds, and print the validity time in seconds.

Reply:
Will do, was going back and forth on which would be easier to read, in case we allowed longer validity in the future